src/Native/String.js

```diff
@@ -29,7 +29,7 @@ function concat(strs)
 }
 function length(str)
 {
-	return str.length;
+	return Array.from(str).length;
```

Comment:
Apparently the same bug in available in Java also.
However, instead of simply making the length linear there should be some lazy, memoized solution (amortized constant time), falling back to `str.length`